R/validate_datalong.R

```diff
@@ -98,11 +98,27 @@ validate_datalong_types <- function(data, vars) {
         msg = sprintf("Variable `%s` should be of type factor", vars$group)
     )
 
+    assert_that(
+        length(unique(data[[vars$group]])) == nlevels(data[[vars$group]]),
+        msg = sprintf(
+            "The number of levels in variable `%s` is different than the number of observed levels",
+            vars$group
+        )
+    )
+
```

Comment:
@gowerc Do you like something like this to check that the number of levels of `group` are the same as the observed levels? Or would you rather check that `unique(data[[vars$group]])) == levels(data[[vars$group]]`?

Reply:
I think what you have here already is fine.